src/app.rs

```diff
@@ -30,10 +31,11 @@ pub struct App<Data> {
 impl<Data: Clone + Send + Sync + 'static> App<Data> {
     /// Set up a new app with some initial `data`.
     pub fn new(data: Data) -> App<Data> {
+        let logger = RootLogger::new();
         App {
             data,
             router: Router::new(),
-            middleware: Vec::new(),
+            middleware: vec![Box::new(logger)],
```

Comment:
Let's hold off on setting up default middleware just yet -- maybe open an issue on it so we can discuss how we want to approach it?